lib/db.js

```diff
@@ -138,9 +138,9 @@ const legalOptionNames = [
  * @fires Db#fullsetup
  * @return {Db} a Db instance.
  */
-function Db(databaseName, topology, options) {
+function Db(databaseName, client, options) {
   options = options || {};
-  if (!(this instanceof Db)) return new Db(databaseName, topology, options);
+  if (!(this instanceof Db)) return new Db(databaseName, client, options);
   EventEmitter.call(this);
 
```

Comment:
I recommend that we pull out the topology here, and assert that it exists. We use it a few places below, and actually call `.on` on it after storing it, so our current requirement already is that it must exist:
```
if (client == null || (client && client.topology == null)) {
  throw MongoError("A client is required to create a Db instance"); // probably some much better error
}

const topology = client.topology;
...
this.s = {
  ...
  topology,
  ...
};

...
topology.on('whatever'...);
```